### PR TITLE
doc: add macOS instructions

### LINKED ISSUE #1450: Add macOS server setup doc

### PR DESCRIPTION
**Pull request by Levilian**
Fix: #1450

### CODE REVIEW

File: docs/Server-Setup.md [hunk 1]
```diff
@@ -83,6 +83,17 @@ Start the CodaLab service as follows:
 
     ./codalab_service.py start -bd
 
+If your default python version is not python 3.6+, then you might have to use:
+
+    python3 codalab_service.py start -bd
```
Comment (percyliang) [inline docs/Server-Setup.md]: Don't need this since `python3.6` is specified in `codalab_service.py` already.

File: docs/Server-Setup.md [hunk 2]
```diff
@@ -83,6 +83,17 @@ Start the CodaLab service as follows:
...
+
+For macOS, you might come across an error with `gunicorn` when running the above command.
```
Comment (percyliang) [inline docs/Server-Setup.md]: Maybe this should be put in a section at the end called Troubleshooting so the main instructions don't get too long.

**Review by percyliang — CHANGES_REQUESTED** [review]

File: docs/Server-Setup.md (hunk 1 above)
Comment (Levilian) [reply docs/Server-Setup.md]: I still receive the following error if I just run ./codalab_service.py
`bash: ./codalab_service.py: /usr/bin/python3: bad interpreter: No such file or directory`

Comment (percyliang) [reply docs/Server-Setup.md]: replace the first line with `#!/usr/bin/env python3`

Comment (Levilian) [reply docs/Server-Setup.md]: Is it because I used anaconda instead of pip to install my python3?

Comment (Levilian) [reply docs/Server-Setup.md]: Yes I originally changed the line to `#!/usr/bin/env python3` but then I don't know if I should update the master branch with this change. I think `#!/usr/bin/env python3` is preferred in shebang as it doesn't assume that it's running on a Ubuntu system. Any reason why we chose to use `#!/usr/bin/python3`?

Comment (percyliang) [reply docs/Server-Setup.md]: It should be `#!/usr/bin/env python3`

Comment (Levilian) [reply docs/Server-Setup.md]: Ok I will create a separate PR and fix that!

File: docs/Server-Setup.md [hunk 3]
```diff
@@ -6,8 +6,8 @@ CodaLab features.
 # Requirements
 
 All you need is a version of Docker and docker-compose compatible with docker-compose file version 3.5.
-Note that while Docker/docker-compose should be cross-platform, we have only tested Ubuntu.
-If you try to use CodaLab on MacOS or Windows and run into issues, feel free to let us know, but those platforms
+Note that while Docker/docker-compose should be cross-platform, we have only tested Ubuntu and MacOS.
```
Comment (epicfaace) [inline docs/Server-Setup.md:9]: macOS

Comment (Levilian) [reply docs/Server-Setup.md:9]: Nice catch

**Pull request merged**